pallets/parachain-staking/src/lib.rs

```diff
@@ -730,8 +733,9 @@ pub mod pallet {
 					.expect("infinite length input; no invalid inputs for type; qed"),
 				percent: self.parachain_bond_reserve_percent,
 			});
-			// Set total selected candidates to minimum config
-			<TotalSelected<T>>::put(T::MinSelectedCandidates::get());
+			// Set total selected candidates to value from config
+			<Pallet<T>>::set_total_selected(T::RuntimeOrigin::root(), self.num_selected_candidates)
```

Comment:
You can do `<TotalSelected<T>>::put(self.num_selected_candidates);`. This will only run at genesis, so there is no previous value nor the need to deposit a event, so I think we can just set the value directly

Reply:
I would need to duplicate this validation, right?

    ensure!(
      new >= T::MinSelectedCandidates::get(),
      Error::<T>::CannotSetBelowMin
    );

I decided to call the function in case the function logic changes in the future, but it is true that some checks are redundant in that case.

Reply:
You could make an `inner_set_total_selected` function which does the validation that it's greater than `MinSelectedCandidates` but does not emit the event. Then call this function in genesis here and in the `set_total_selected` extrinsic.

> I decided to call the function in case the function logic changes in the future, but it is true that some checks are redundant in that case.

This makes sense and I'd prefer something that enforces the same checks in both places instead of just duplicating code.

Reply:
Else you can just take the minimum between totalSelected and MinSelectedCandidates for the genesis

Reply:
I decided to simply duplicate the validation, as splitting the function in 2 did not feel clean. And I prefer an explicit panic instead of silently using a different value.